pest/src/error.rs

```diff
@@ -524,6 +540,43 @@ fn visualize_whitespace(input: &str) -> String {
     input.to_owned().replace('\r', "␍").replace('\n', "␊")
 }
 
+#[cfg(feature = "miette")]
+mod miette {
+    use alloc::string::ToString;
+    use std::boxed::Box;
+
+    use super::{Error, InputLocation, RuleType};
+
+    use miette::{Diagnostic, LabeledSpan, SourceCode};
+
+    #[derive(thiserror::Error, Debug)]
+    #[error("Failure to parse at {}", self.0.line_col)]
+    pub(crate) struct MietteAdapter<R: RuleType>(pub(crate) Error<R>);
+
+    impl<R: RuleType> Diagnostic for MietteAdapter<R> {
```

Comment:
errors could also contain `path` and `continued_line` -- should those also be included in a `Diagnostic` (e.g. as additional labels) or would including those make the Miette default formatted print outs look strange?

Reply:
Yes, it would make the miette-fornatted output strange. Adding it to the Display is fine, though.